src/transformers/models/bart/modeling_bart.py

```diff
@@ -1036,7 +1040,7 @@ def forward(
             encoder_attention_mask = _expand_mask(encoder_attention_mask, inputs_embeds.dtype, tgt_len=input_shape[-1])
 
         # embed positions
-        positions = self.embed_positions(input_shape, past_key_values_length)
+        positions = self.embed_positions(input, past_key_values_length)
```

Comment:
Looks like it could always be `inputs_embeds[:, :, -1]` here since `input_embeds` are now defined.

Reply:
Perhaps I'm missing something, but if `inputs_embeds` is not provided, the `input` here will be `input_ids`. If `inputs_embeds` _is_ provided (but `input_ids` is not), this `input` will be `inputs_embeds[:, :, -1]`. So the dependency on what is provided means we cannot always pass `inputs_embeds[:, :, -1]`.

Reply:
Yes but the `input_embeds` are now defined and their shape will match. Anyhow it's not important, it was to go along with the suggestion above to remove `input` entirely.